docs/rules/consistent-return.md

```diff
@@ -77,6 +71,95 @@ function Foo() {
 }
 ```
 
+## Options
+
+This rule has an object option:
+
+* `"treatUndefinedAsUnspecified": false` (default) always either specify values or return `undefined` implicitly only.
+* `"treatUndefinedAsUnspecified": true` always either specify values or return `undefined` explicitly or implicitly.
+
+### treatUndefinedAsUnspecified
+
+Examples of **incorrect** code for this rule with the default `{ "treatUndefinedAsUnspecified": false }` option:
+
+```js
+/*eslint consistent-return: ["error", { "treatUndefinedAsUnspecified": false }]*/
+
+function foo(callback) {
+    if (callback) {
+        return void callback();
+    }
+    // no return statement
+}
+
+function bar(condition) {
+    if (condition) {
+        return undefined;
+    }
+    // no return statement
+}
+```
+
+Examples of **correct** code for this rule with the default `{ "treatUndefinedAsUnspecified": false }` option:
```

Comment:
Could we omit these examples of **correct** for default option because they are also **incorrect** for new option?

Not because of the repetition itself, but because they look like an anti-pattern for the goal of this rule.